### ISSUE TITLE
2.4.2 causing core panic in ESP32

### ISSUE THREAD

**Issue by willemneethling** [Bodmer/TFT_eSPI#1574]
Firstly, thank you for a great library and all your hard work. I recently tried to use this library again on an ESP32 and encountered issues using the latest version, but not the version before that.

_Hardware:_ 
ESP32 with a 3.2 TFT 240x320, no touch, ILI9341_DRIVER

_Steps to reproduce:_ 
Run any TFT_eSPI example on an ESP32 using version 2.4.2

Using version **2.4.2** causes a core panic on an ESP32:

```
15:46:06.910 -> Rebooting...
15:46:06.910 -> ets Jul 29 2019 12:21:46
15:46:06.910 -> 
15:46:06.910 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
15:46:06.910 -> configsip: 0, SPIWP:0xee
15:46:06.910 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
15:46:06.910 -> mode:DIO, clock div:1
15:46:06.910 -> load:0x3fff0018,len:4
15:46:06.910 -> load:0x3fff001c,len:1216
15:46:06.910 -> ho 0 tail 12 room 4
15:46:06.910 -> load:0x40078000,len:9720
15:46:06.910 -> ho 0 tail 12 room 4
15:46:06.910 -> load:0x40080400,len:6352
15:46:06.910 -> entry 0x400806b8
15:46:07.051 -> Guru Meditation Error: Core  1 panic'ed (IllegalInstruction). Exception was unhandled.
15:46:07.097 -> Memory dump at 0x400d4da0: 01810004 800118d4 d10c0800
15:46:07.097 -> Core 1 register dump:
15:46:07.097 -> PC      : 0x400d4da4  PS      : 0x00060530  A0      : 0x800d1adc  A1      : 0x3ffb1f10  
15:46:07.097 -> A2      : 0x3ffbec04  A3      : 0x00000017  A4      : 0x00000001  A5      : 0x00000000  
15:46:07.097 -> A6      : 0x3ffbec04  A7      : 0x00000002  A8      : 0x00000000  A9      : 0x3ffb1ed0  
15:46:07.097 -> A10     : 0x00000001  A11     : 0x00000000  A12     : 0x3ffb8484  A13     : 0x00000000  
15:46:07.097 -> A14     : 0x00000000  A15     : 0x00000004  SAR     : 0x0000000d  EXCCAUSE: 0x00000000  
15:46:07.145 -> EXCVADDR: 0x00000000  LBEG    : 0x400d5204  LEND    : 0x400d5266  LCOUNT  : 0x00000002  
15:46:07.145 -> 
15:46:07.145 -> Backtrace: 0x400d4da4:0x3ffb1f10 0x400d1ad9:0x3ffb1f30 0x400d222f:0x3ffb1f50 0x400d0dc1:0x3ffb1f80 0x400d5db7:0x3ffb1fb0 0x400897e5:0x3ffb1fd0
```
Using exactly the same board and the same User_Setup.h, but version **2.3.70** of the library works fine. I tested on a ESP32-Devkit-C as well as a Wemos D32. The core panic happens on any sketch, even the Read_User_Setup sketch

Here is the relevant output from Read_User_Setup using version 2.3.70:

```
15:29:31.179 -> TFT_eSPI ver = 2.3.70
15:29:31.179 -> Processor    = ESP32
15:29:31.179 -> Frequency    = 240MHz
15:29:31.179 -> Transactions = Yes
15:29:31.179 -> Interface    = SPI
15:29:31.179 -> Display driver = 9341
15:29:31.179 -> Display width  = 240
15:29:31.223 -> Display height = 320
15:29:31.223 -> 
15:29:31.223 -> MOSI    = GPIO 23
15:29:31.223 -> MISO    = GPIO 19
15:29:31.223 -> SCK     = GPIO 18
15:29:31.223 -> TFT_CS   = GPIO 15
15:29:31.223 -> TFT_DC   = GPIO 2
15:29:31.223 -> TFT_RST  = GPIO 4
15:29:31.223 -> 
15:29:31.223 -> Font GLCD   loaded
15:29:31.223 -> Font 2      loaded
15:29:31.223 -> Font 4      loaded
15:29:31.223 -> Font 6      loaded
15:29:31.223 -> Font 7      loaded
15:29:31.223 -> Font 8      loaded
15:29:31.223 -> Smooth font enabled
15:29:31.223 -> 
15:29:31.223 -> Display SPI frequency = 27.00
```

Are there any breaking changes in version 2.4.2 that needed other changes to the User_Setup.h file, or is this a bug?

**Comment by Bodmer** [comment]
I think this is down to board package changes. Can you try the latest 2.0.2 ESP32 board package.

![image](https://user-images.githubusercontent.com/15824805/150862974-ad99d53d-cb84-49d8-8826-6db8cfbe3401.png)

**Comment by willemneethling** [comment]
Thanks for the prompt feedback. I neglected to add my esp32 boards version. I was on the master release so I did not get version 2.0.2, but switching to the development release, I could install it and tested. 

Library 2.4.2 is working fine on esp32 boards 2.0.2 

For anyone else facing this issue, if you are using the master releases for the esp33 boards, use a lower version of TFT_eSPI, e.g.  2.3.70

If you are using the development release (>= 2.0.2) of esp32 boards then TFT_eSPI v. 2.4.2 works fine

This can be closed, as its not a bug
